src/command/archive/get/file.c

```diff
@@ -53,24 +89,13 @@ archiveGetFile(
                 StorageWrite *const destination = storageNewWriteP(
                     storage, walDestination, .noCreatePath = true, .noSyncFile = true, .noSyncPath = true, .noAtomic = true);
 
-                // If there is a cipher then add the decrypt filter
-                if (actual->cipherType != cipherTypeNone)
-                {
-                    ioFilterGroupAdd(
-                        ioWriteFilterGroup(storageWriteIo(destination)),
-                        cipherBlockNewP(cipherModeDecrypt, actual->cipherType, BUFSTR(actual->cipherPassArchive)));
-                    compressible = false;
-                }
+                compressible = buildArchiveGetPipeLine(ioWriteFilterGroup(storageWriteIo(destination)), actual);
 
-                // If file is compressed then add the decompression filter
-                CompressType compressType = compressTypeFromName(actual->file);
-
-                if (compressType != compressTypeNone)
+                if (walIsSegment(request) && cfgOptionTest(cfgOptFilter))
```

Comment:
Both of these conditions are independent of the loop, but they contain many function calls. Therefore, I suggest calculating them before the loop into a Boolean variable, and only checking it here.